dtox.sh

```diff
@@ -7,40 +7,59 @@ ROOT="$(git rev-parse --show-toplevel)"
 BASE_INPUT=(
   "${ROOT}/docker/base/Dockerfile"
 )
-base_hash=$(cat ${BASE_INPUT[@]} | git hash-object -t blob --stdin)
+base_hash=$(cat "${BASE_INPUT[@]}" | git hash-object -t blob --stdin)
 
 function base_id() {
-  docker images -q -f label=base_hash=${base_hash} pantsbuild/pex:base
+  docker images -q -f label=base_hash="${base_hash}" pantsbuild/pex:base
 }
 
 if [[ -z "$(base_id)" ]]; then
   docker build \
     --tag pantsbuild/pex:base \
-    --label base_hash=${base_hash} \
+    --label base_hash="${base_hash}" \
     "${ROOT}/docker/base"
 fi
 
 USER_INPUT=(
   "${ROOT}/docker/user/Dockerfile"
   "${ROOT}/docker/user/create_docker_image_user.sh"
 )
-user_hash=$(cat ${USER_INPUT[@]} | git hash-object -t blob --stdin)
-if [[ -z "$(docker images -q -f label=user_hash=${user_hash} pantsbuild/pex:user)" ]]; then
+user_hash=$(cat "${USER_INPUT[@]}" | git hash-object -t blob --stdin)
+if [[ -z "$(docker images -q -f label=user_hash="${user_hash}" pantsbuild/pex:user)" ]]; then
   docker build \
-    --build-arg BASE_ID=$(base_id) \
-    --build-arg USER=$(id -un) \
-    --build-arg UID=$(id -u) \
-    --build-arg GROUP=$(id -gn) \
-    --build-arg GID=$(id -g) \
+    --build-arg BASE_ID="$(base_id)" \
+    --build-arg USER="$(id -un)" \
+    --build-arg UID="$(id -u)" \
+    --build-arg GROUP="$(id -gn)" \
+    --build-arg GID="$(id -g)" \
     --tag pantsbuild/pex:user \
-    --label user_hash=${user_hash} \
+    --label user_hash="${user_hash}" \
     "${ROOT}/docker/user"
 fi
 
+DOCKER_ARGS=()
+if [[ "$1" == "inspect" ]]; then
+  shift
+  DOCKER_ARGS+=(
+    --entrypoint bash
+  )
+fi
+
 exec docker run \
-  --interactive \
-  --tty \
-  --rm \
-  --volume $(pwd):/dev/pex \
-  pantsbuild/pex:user \
-  "$@"
+    --interactive \
+    --tty \
+    --rm \
+    --volume "${HOME}/.netrc:/home/$(id -un)/.netrc" \
+    --volume "${HOME}/.ssh:/home/$(id -un)/.ssh" \
```

Comment:
These are needed for various ITs involving netrc support and GH repo cloning (VCS requirement tests).